### PR TITLE
remove cookie handling

### PR DESCRIPTION
**Pull request by darkonie**
DCOS-12011

### CODE REVIEW

File: dcos-log/api/middleware.go [hunk 1]
```diff
@@ -63,39 +63,19 @@ func validateToken(t string) (string, error) {
...
 
-// getAuthOrCookieFromRequest will try to extract JWT from Authorization header first. If it is not there
+// getAuthFromRequest will try to extract JWT from Authorization header first. If it is not there
 // it will try to look for a dcos-acs-auth-cookie.
-func getAuthOrCookieFromRequest(r *http.Request) (string, error) {
+func getAuthFromRequest(r *http.Request) (string, error) {
 	// give priority to Authorization header
 	authorizationHeader := r.Header.Get("Authorization")
 	if authorizationHeader != "" {
 		return validateToken(authorizationHeader)
```
Comment (malnick) [inline dcos-log/api/middleware.go]: should we provide this validation in the dcos-go library?

File: dcos-log/api/middleware.go [hunk 2]
```diff
@@ -63,39 +63,19 @@ func validateToken(t string) (string, error) {
...
-	}
-
-	// TODO: improve token validation
+	
 	return t, nil
 }
 
-// getAuthOrCookieFromRequest will try to extract JWT from Authorization header first. If it is not there
+// getAuthFromRequest will try to extract JWT from Authorization header first. If it is not there
 // it will try to look for a dcos-acs-auth-cookie.
```
Comment (jgehrcke) [inline dcos-log/api/middleware.go]: This comment is now stale @darkonie

File: dcos-log/api/middleware.go [hunk 3]
```diff
@@ -63,39 +63,19 @@ func validateToken(t string) (string, error) {
 	if !strings.HasPrefix(t, "token=") {
 		return t, ErrInvalidToken
```
Comment (jgehrcke) [inline dcos-log/api/middleware.go]: This should probably be `ErrMissingToken`, right? What we detect here is the case when no token is present.

`ErrInvalidToken` (and the error message "Invalid token used") must only be used when the token is actually, intrinsically, invalid. Which we do not check here. So it seems like `ErrInvalidToken` can be removed from the code entirely.

File: dcos-log/api/middleware.go (hunk 2 above)
Comment (darkonie) [inline dcos-log/api/middleware.go]: fixed. Thanks!

File: dcos-log/api/middleware.go (hunk 3 above)
Comment (darkonie) [inline dcos-log/api/middleware.go]: fixed, thanks!

**Pull request merged**